new/analyse/determine-version.ts

```diff
@@ -0,0 +1,53 @@
+import type { Module, PackageAfterDetermineVersion } from '../types';
+
+import { logger } from '../helpers/logger';
+import { clone } from 'ramda';
+import semver from 'semver';
+
+const { error, log } = logger('determine Version (1)');
+
+export const determineVersion: Module = (env) => {
+  if (!env.packages) {
+    const errorMessage = '"packages" not present on env config object.';
+    error(errorMessage);
+    throw new Error(errorMessage);
+  }
+
+  log('analyse packages');
+
+  const clonePackages = clone(env.packages);
+
+  const incrementLevels: semver.ReleaseType[] = ['patch', 'minor', 'major'];
+
+  Object.keys(clonePackages).forEach((packageName) => {
+    const { determinedIncrementLevel, packageJson } =
+      clonePackages[packageName];
+
```

Comment:
Somewhere around here should be a check if `determinedIncrementLevel !== -1` because only in this case, we need to do something.

Reply:
I'll add it